### ISSUE TITLE
(A1) path traversel lesson 7 seems broken

### ISSUE THREAD

**Issue by shoaloak** [WebGoat/WebGoat#1103]
The zip slip assignment appears to be broken.

The goal is to somehow override `/WebGoat/images/account.png` when looking at the requests.
The current location where the zip is extracted is `/home/webgoat/.webgoat-8.2.1/PathTraversal/<NAME>/<INJECT_ME/FILENAME>.png`

Uploading a zip with just containing an image, e.g., `hack.png`, marks the lesson as completed.
However, the info message is `Zip file extracted successfully, failed to copy image. Please contact our helpdesk.`

It is also unclear from inspecting the Docker container where the image should be put. The source seems to always call `getProfilePicture`, which fetches it from `this.webGoatHomeDirectory, "/PathTraversal/" + webSession.getUserName()`, which is where the image is already extracted?

**Comment by mnemo9** [comment]
I experience the same issue with webgoat-8.2.2

> The goal is to somehow override /WebGoat/images/account.png

I don't think it is. That file is not in the profile directory (~/.webgoat-8.2.2) but part of the webserver.

I have found this seemingly-related function which weirdly enough doesn't seem to get called at all:

```js
webgoat.customjs.profileZipSlipRetrieval = function () {
    $.get("PathTraversal/zip-slip", function (result, status) {
        document.getElementById("previewZipSlip").src = "data:image/png;base64," + result;
    });
}
```

The question remains - where should the uploaded picture go to ? The challenge is marked as finished sending any zip file.

Page 8 of the lesson shows that the solution is about overwriting:
```
.webgoat-8.2.2/PathTraversal/testuser/testuser.jpg
```
but uploading that didn't work for me and the profile picture remained the same.

Also the java fix code shown in the solution seems abrupt:
```java
File profilePicture = new File(uploadDirectory, e.getName());
if (profilePicture.
```

**Comment by nbaars** [comment]
> One of these is that you are required to have passed lesson 2 before lesson 7.

That is related to the reuse of the same directory. The Zip Slip lesson now cleans the directory before processing the upload of the zipfile.

**Comment by nbaars** [comment]
Closing as we released 2023.3
